### PR TITLE
SuppressWarnings for ExcessivePublicCount #409

### PR DESCRIPTION
**Pull request by eeree**
I've checked if phpmd/phpmd#409 is still valid and it looks it's not. I've covered it with tests to prove it. Issue may be closed.
Closes phpmd/phpmd#409

### CODE REVIEW

**Comment by ravage84** [comment]
Confirmed in PhpStorm: Code taken from your PR, the four violation messages are without suppression, the green status message is with all four supressions.

File: src/test/php/PHPMD/Regression/ExcessivePublicCountWorksCorrectlyWithStaticMethodsTest.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                $this->returnCallback(
+                    function (Report $report) use ($self) {
+                        $isViolating = false;
+                        foreach ($report->getRuleViolations() as $ruleViolation) {
+                            if (strpos($ruleViolation->getDescription(), $self::VIOLATION_MESSAGE) === 0) {
+                                $isViolating = true;
+                                break;
+                            }
+                        }
+                        $self->assertTrue($isViolating);
```
Comment (ravage84) [inline src/test/php/PHPMD/Regression/ExcessivePublicCountWorksCorrectlyWithStaticMethodsTest.php]: PHP 5.3 compat issue persists
https://travis-ci.org/phpmd/phpmd/jobs/243208735#L254

Comment (eeree) [inline src/test/php/PHPMD/Regression/ExcessivePublicCountWorksCorrectlyWithStaticMethodsTest.php]: Not anymore https://travis-ci.org/phpmd/phpmd/builds/243211521

Comment (ravage84) [inline src/test/php/PHPMD/Regression/ExcessivePublicCountWorksCorrectlyWithStaticMethodsTest.php]: Odd. GitHub linked to an outdated build... Never mind.

**Pull request merged**